File: app/src/main/java/lt/vilnius/tvarkau/utils/TextUtils.java
package lt.vilnius.tvarkau.utils;

import java.util.ArrayList;
import java.util.List;
import java.util.regex.Matcher;
import java.util.regex.Pattern;

import static java.util.regex.Pattern.CASE_INSENSITIVE;

/**
 * @author Martynas Jurkus
 */

public class TextUtils {

    private final static String PROBLEM_ID_REGEXP = ("E50-\\S+?\\)");
    private final static Pattern PROBLEM_ID_PATTERN = Pattern.compile(PROBLEM_ID_REGEXP, CASE_INSENSITIVE);


    private TextUtils() {
    }

    public static List<String> findProblemIdOccurrences(String source) {
        List<String> result = new ArrayList<>();
        Matcher matcher = PROBLEM_ID_PATTERN.matcher(source);

        while (matcher.find()) {
            result.add(matcher.group());
        }

        return result;
    }
}

File: app/src/main/java/lt/vilnius/tvarkau/utils/TextUtils.kt
package lt.vilnius.tvarkau.utils

import kotlin.text.RegexOption.IGNORE_CASE

object TextUtils {

    private val PROBLEM_ID_REGEXP = """[A-Z]\d{2}-\d+/\d+\(\S+?\)""".toRegex(IGNORE_CASE)

    fun findReportREferenceNumberOccurrences(source: String): List<String> {
        return PROBLEM_ID_REGEXP.findAll(source).map { it.value }.toList()
    }
}
